FinSearchUnified/tests/Components/ProductStream/CriteriaFactoryTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace FinSearchUnified\tests\Components\ProductStream;
+
+use FinSearchUnified\Constants;
+use Shopware\Components\Test\Plugin\TestCase;
+
+class CriteriaFactoryTest extends TestCase
+{
+    /**
+     * Provider for test cases to test CriteriaFactory
+     *
+     * @return array
+     */
+    public function providers()
+    {
+        return [
+            "useShopSearch returns true" => [
```

Comment:
- Please always use single quotes **unless** you have a good reason to use double quotes e.g.
```
$str = "Shopkey is 'Findologic ShopKey'";
```

- Rename this to `Uses the original implementation`.